app-typescript/components/_Positioner.tsx

```diff
@@ -100,28 +104,41 @@ class PopperWrapper extends React.Component<IPropsPopperWrapper> {
                         this.props.close();
                     }
                 }}
+                onMouseEnter={this.props.onMouseEnter}
+                onMouseLeave={this.props.onMouseLeave}
             >
                 {this.props.children}
             </div>
         );
     }
 }
 
-interface IPropsPositioner {
+interface IBasePositioner {
     triggerSelector: string;
     placement: PopperOptions['placement'];
     className?: string;
 }
 
+interface IPropsPositioner extends IBasePositioner {
+    type: 'click';
+}
+
+interface IPropsPositionerOnHover extends IBasePositioner {
+    type: 'hover';
+}
+
+type IProps = IPropsPositioner | IPropsPositionerOnHover;
+
```

Comment:
I think this can be simplified. Unless there is another reason to have two separate interfaces.
```suggestion
interface IProps extends IBasePositioner {
    type: 'click' | 'hover';
}

```

Reply:
rather a good practice, leave room open when different props apply to different types. We showcase that it can be type narrowed

Reply:
but do you think `IPropsPositioner` and `IPropsPositionerOnHover` will be extended further from what you have there?